### PR TITLE
BaSP-TG-54: Fix getById method for Admins entity

### PR DESCRIPTION
**Pull request by JuliMonte5**
1. getById method was returning status 200 for non-existent Ids in the DB
2. Also included a validation for a correct Id format (to be a mongoose type)

### CODE REVIEW

File: src/controllers/admins.js [hunk 1]
```diff
@@ -1,17 +1,35 @@
+import mongoose from 'mongoose';
 import Admins from '../models/Admins';
 
 const getAdminById = async (req, res) => {
+  if (req.params.id && !mongoose.Types.ObjectId.isValid(req.params.id)) {
+    return res.status(400).json({
+      message: `Cannot get Admin by ${req.params.id}`,
```
Comment (Ignacio-Sormani) [inline src/controllers/admins.js:7]: The message should be something like:
`${req.params.id} is an invalid id`

File: src/controllers/admins.js [hunk 2]
```diff
@@ -1,17 +1,35 @@
...
+      data: undefined,
+      error: true,
+    });
+  }
   try {
     const { id } = req.params;
```
Comment (Ignacio-Sormani) [inline src/controllers/admins.js:13]: Move this to the first line of the function

File: src/controllers/admins.js [hunk 3]
```diff
@@ -1,17 +1,35 @@
...
-      data: admin,
+    if (admin) {
+      return res.status(200).json({
+        message: 'Admin Found',
+        data: admin,
+        error: false,
+      });
+    }
+    return res.status(404).json({
+      message: `Admin with id ${req.params.id} not found`,
```
Comment (Ignacio-Sormani) [inline src/controllers/admins.js:23]: Just use id const

**Review by Ignacio-Sormani — CHANGES_REQUESTED** [review]

File: src/controllers/admins.js (hunk 1 above)
Comment (Ignacio-Sormani) [reply src/controllers/admins.js:7]: This was resolved in https://github.com/BaSP-a2022/ayom-b-trackgenix-server/pull/57

**Review by GonzaloAsencio — APPROVED** [review]

**Review by Ignacio-Sormani — APPROVED** [review]

**Review by sofihoj — APPROVED** [review]

**Pull request merged**